### PR TITLE
Generate OpenAPI specification for links endpoint

### LINKED ISSUE #3153: Add OpenAPI Specification for /links/ endpoint
## Problem
To ensure the accuracy of the specs generated by drf-spectacular, we will generate an OpenAPI Specification specifically for a particular endpoint. This will allow us to verify its correctness by comparing it with the existing test cases and the DRF browsable API page. This issue aims to generate spec for /links/ endpoint.

## Proposed solution
* Add /links/ to the custom preprocessing hook function to filter out all endpoints except for the /links/ endpoint. 
* Generate the spec file using the _./manage.py spectacular --color --file schema.yml_ command

### PR DESCRIPTION
**Pull request by varshadr**
Fixes #3153

This PR add openAPI specification for `links` endpoint to the schema file

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: schema.yml [hunk 1]
```diff
@@ -266,6 +374,33 @@ paths:
               schema:
                 $ref: '#/components/schemas/Database'
           description: ''
+  /api/db/v0/links/:
```
Comment (varshadr) [inline schema.yml:377]: @silentninja could you please review the spec for `links`, here?

**Review by silentninja — APPROVED** [review]

**Review by silentninja — APPROVED** [review]
Looks good to me! @varshadr Can you resolve the conflicts and merge the pull request

File: schema.yml (hunk 1 above)
Comment (varshadr) [reply schema.yml:377]: @silentninja I'm suggesting we close this PR. The `/links/` endpoint has already been merged into the develop branch and its API documentation is generated through another PR that is merged.

**Comment by seancolsen** [comment]
@varshadr, in looking over PRs, I've noticed a few assigned to Mukesh. We shouldn't have any PRs assigned to him anymore. In general, please assign to @mathemancer going forward.

But for this PR in particular, I'm not sure why you've assigned it to anyone else, given the recent approving review post here. As far as I can tell, the next steps for this PR are for you to resolve the merge conflicts and then merge the PR yourself. If you need help with any remaining steps, please ask specific questions to specific people so that the comments here clearly indicate the remaining work needed. If you need help resolving merge conflicts, it might be good to ask in Matrix. One of us can even hop on a call with you to help out if needed.

**Comment by varshadr** [comment]
> @varshadr, in looking over PRs, I've noticed a few assigned to Mukesh. We shouldn't have any PRs assigned to him anymore. In general, please assign to @mathemancer going forward.
> 
> But for this PR in particular, I'm not sure why you've assigned it to anyone else, given the recent approving review post here. As far as I can tell, the next steps for this PR are for you to resolve the merge conflicts and then merge the PR yourself. If you need help with any remaining steps, please ask specific questions to specific people so that the comments here clearly indicate the remaining work needed. If you need help resolving merge conflicts, it might be good to ask in Matrix. One of us can even hop on a call with you to help out if needed.

@seancolsen The `/links/` endpoint spec is already present in the develop branch, having been merged through another PR. Generating API documentation for each API endpoint individually was initially helpful for reviewing the spec more clearly. However, the spec has now been reviewed by Mukesh and looks fine.
Therefore, I'd want to go ahead and close this PR. What do you think @seancolsen and @mathemancer?

**Comment by mathemancer** [comment]
We're reprioritizing, and  unfortunately we won't be continuing the documentation project.

**Pull request closed without merging**